src/WebJobs.Extensions.DurableTask/DurableHttpClientFactory.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class DurableHttpClientFactory
+    {
+        private HttpClient durableHttpClient;
+
+        public HttpClient GetClient(IDurableHttpMessageHandlerFactory handler)
+        {
+            if (this.durableHttpClient == null)
+            {
+                this.durableHttpClient = new HttpClient(handler.CreateHttpMessageHandler());
+
+                var assembly = typeof(DurableTaskExtension).Assembly;
+                Version assemblyVersion = GetAssemblyVersion();
+
+                this.durableHttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(
+                    assembly.GetName().Name,
+                    assemblyVersion.ToString()));
+            }
+
+            return this.durableHttpClient;
+        }
+
+        private static Version GetAssemblyVersion()
+        {
+            var assembly = typeof(DurableTaskExtension).Assembly;
```

Comment:
This call to `typeof(DurableTaskExtension).Assembly` is redundant to the code from a few lines above. I suggest you pass in the `Assembly` reference as a parameter.